File: problem026/solution.php
<?php

require __dir__ . "/../src/helpers.php";

$start = microtime(true);

function coprime10($x)
{
    return $x % 2 && $x % 5;
}

function mOrder(int $test)
{
    for ($i = 1; $i <= 1000; $i++) {
        if (bcmod(bcpow(10, $i), $test) == 1) {
            return $i;
        }
    }
}

$solution = 0;
$max = 0;

for ($x = 2; $x < 1000; $x++) {
    if (coprime10($x)) {
        $m = mOrder($x);
        if ($m > $max) {
            $solution = $x;
            $max = $m;
        }
    }
}

solution($solution);
duration($start);

File: problem026/solution.py
def coprime10(x):
    return x % 2 and x % 5


def order(x):
    for i in range(1, 1000):
        if (10 ** i) % x == 1:
            return int(i)


solution = 0
maximum = 0

for i in [i for i in range(2, 1001) if coprime10(i)]:
    m = order(i)
    if m > maximum:
        solution = i
        maximum = m

print(solution)
